src/components/settings/data/hooks.js

```diff
@@ -60,7 +61,59 @@ export const useLinkManagement = (enterpriseUUID) => {
   };
 };
 
+const initialCustomerAgreementState = {
+  netDaysUntilExpiration: 0,
+};
+/*
+ * @param {Objet {enterpriseId: string}}
+ * @returns {customerAgreement: Object, loadingCustomerAgreement: bool}
+ * customerAgreement:{
+ *  netDaysUntilExpiration: number
+ * }
+ */
+export const useCustomerAgreementData = ({ enterpriseId }) => {
+  const [customerAgreement, setCustomerAgreement] = useState(initialCustomerAgreementState);
+  const [loadingCustomerAgreement, setLoadingCustomerAgreement] = useState(true);
+
+  const loadCustomerAgreementData = (page = 1) => {
+    if (!loadingCustomerAgreement) {
+      setLoadingCustomerAgreement(true);
+    }
+    LicenseManagerApiService.fetchCustomerAgreementData({ enterprise_customer_uuid: enterpriseId, page })
+      .then((response) => {
+        const newCustomerAgreementData = {
+          netDaysUntilExpiration: 0,
+        };
+        const { data: customerAgreementData } = camelCaseObject(response);
+        if (customerAgreementData.results && customerAgreementData.count) {
+          // Only look at customer agreements with subs
+          customerAgreementData.results.filter(result => (result.subscriptions && result.subscriptions.length))
```

Comment:
Interesting. I get why you implemented it this way (i.e., the API response is paginated and it seems your logic may have been inspired by [this code](https://github.com/openedx/frontend-app-admin-portal/blob/master/src/components/subscriptions/data/hooks.js#L28)), but the CustomerAgreement model's `enterprise_customer_uuid` field is [unique](https://github.com/openedx/license-manager/blob/master/license_manager/apps/subscriptions/models.py#L76), so the API call should only return at most 1 `CustomerAgreement` at any given time. This means the iteration you're doing over a list of `CustomerAgreement` objects might be unnecessary.

What are your thoughts about baking in this assumption of at most 1 `CustomerAgreement` into this logic rather than doing the iteration? I guess the reason why we might want to keep this iteration logic in place might be so that we don't need to make any changes to the frontend if/when the `CustomerAgreement` model no longer has a unique constraint of the `enterprise_customer_uuid`. What do you think?

Reply:
I don't feel strongly about this. I did get most of this code from the code you linked and thought about applying the assumption you described... 

But basically just don't like hard coding a zero like this:
```
customerAcgreement = customerAgreementData.results[0]
```
Also gives us some way to reason in the case there is a freak accident and we get more than one `CustomerAgreement`.

If you feel like making the constraint that we always look at the first thing in `customerAgreementData.results` I'd be ok doing so.